agents/services/agents.js

```diff
@@ -8,24 +8,40 @@ const hooks = {
     create: [
       (hook) => {
         if (hook.result.type === 'person') {
+          const agentId = hook.result.id
+          var agentEmail
+
           return hook.app.service('taskPlans').create({
-            assigneeId: hook.result.id,
+            assigneeId: agentId,
             taskRecipeId: 'createProfile',
             params: {}
           })
           .then(() => {
-            return hook.app.service('credentials').get(hook.result.id)
+            return hook.app.service('credentials').find({ query: { agentId } })
+          })
+          .then(([cred]) => {
+            agentEmail = cred.email
+            return hook.app.service('tokens').create({
+              agentId,
+              service: 'credentials',
+              method: 'patch',
+              params: { serviceId: cred.id },
+              createdAt: new Date().toString()
+            })
           })
-          .then((cred) => {
+          .then((token) => {
             // TODO: IK: looks like the email isn't getting attached to the credential correctly, investigate dogstack-agents
             return hook.app.service('mailer').create({
               from: 'hello@cobuy.nz',
-              to: cred.email || 'no@email.com',
-              subject: 'Test Cobuy SignUp Email',
-              html: 'test email'
+              to: agentEmail || 'no@email.com',
```

Comment:
this email should only be sent if you don't have an existing credential (password, oauth login, etc)

Reply:
if you are the first one to create an account (the group admin), we need a separate email flow to verify their email

Reply:
@ahdinosaur keen on your thoughts on this, looking at this problem again today...

so at the moment this is a after-create hook on `agents`, which gives us the problem that this hook fires before the `credentials` service has finished creating a users credential - meaning we can't tell if this is an invited user (through MemberInvites) or a user signing up by themselves.

so I thought aha! why not just a hook on `credentials` rather than `agents`? but after experimenting with that, i learned that currently the flow is:
- registering creates a `credential`, which then creates an `agent` if there is no agent data in the `before` hook
- creating that `agent` then creates a `credential` in the `after` hook (this credential has no data associated with it apart from `agentId`, i.e. no password or Oauth id)
so hooking onto `credentials` for email invites ends up sending 2 invites, which is bad, and would also send an invite for any other credential created for a user, also bad. 

i'm wondering now whether to prevent that second `credential` creation... but i can imagine situations where we want creating an `agent` to also create a `credential`. and that kinda strikes at the point as i'm thinking now - perhaps we should BE creating `agents` which then create `credentials`, rather than the other way around? `MemberInvites` works this way - is there a reason registering doesn't?